modules/template_architecture_definition/data.tf

```diff
@@ -0,0 +1,9 @@
+data "template_file" "populated_architecture_definition_json" {
```

Comment:
Can we change this to use the `template_file` function and remove the reference to the deprecated `template` provider?

This provider is not supported and does not work with macos.